### PR TITLE
Skip zero-byte S3 objects

### LINKED ISSUE #2: inputs/s3.rb: Keep attempting and failing to parse the same gz file if it's corrupted
Moved from https://github.com/elasticsearch/logstash/issues/2090

I'm using logstash v1.4.2. S3 input crashes if a bucket contains a zero-byte gz file. (throwing an error "Error: not in gzip format".) Then, logstash will retry indefinitely since sincedb is not updated.

I've made a change in `list_new` to skip empty S3 objects (attached code below) to address this particular issue but it may be a good idea to have more robust error handling.

``` ruby
  private
  def list_new(since=nil)

    if since.nil?
      since = Time.new(0)
    end

    objects = {}
    @s3bucket.objects.with_prefix(@prefix).each do |log|
      # original code in v1.4.2
      # if log.last_modified > since
      if log.last_modified > since && log.content_size > 0
        objects[log.key] = log.last_modified
      end
    end

    return sorted_objects = objects.keys.sort {|a,b| objects[a] <=> objects[b]}

  end # def list_new
```

### PR DESCRIPTION
**Pull request by shuwada**
Fix #2.

Hi, I'm the one originally reported the issue. The issue still remains in master and it would be appreciated if you could merge this patch into the mainline.

### CODE REVIEW

**Comment by shuwada** [comment]
CLA has been signed. I'll resubmit this PR if it's necessary.

File: lib/logstash/inputs/s3.rb [hunk 1]
```diff
@@ -111,7 +111,7 @@ def list_new_files
       @logger.debug("S3 input: Found key", :key => log.key)
 
       unless ignore_filename?(log.key)
-        if sincedb.newer?(log.last_modified)
+        if sincedb.newer?(log.last_modified) && log.content_size > 0
```
Comment (ph) [inline lib/logstash/inputs/s3.rb]: If I read the documentation carefully it should be `content_length`
http://docs.aws.amazon.com/AWSRubySDK/latest/AWS/S3/S3Object.html#content_length-instance_method

**Comment by watollop** [comment]
Do we have any news on this?

**Comment by jsvd** [comment]
this has been solved in https://github.com/logstash-plugins/logstash-input-s3/commit/fe49a605638a28e522b93718a31d1ed08365a084

**Pull request closed without merging**